components/script/dom/htmlformelement.rs

```diff
@@ -253,6 +262,214 @@ impl HTMLFormElementMethods for HTMLFormElement {
         let elements = self.Elements();
         elements.IndexedGetter(index)
     }
+
+    // https://html.spec.whatwg.org/multipage/#the-form-element%3Adetermine-the-value-of-a-named-property
+    fn NamedGetter(&self, name: DOMString) -> Option<RadioNodeListOrElement> {
+        let mut candidates: Vec<DomRoot<Node>> = Vec::new();
+
+        let controls = self.controls.borrow();
+        // Step 1
+        for child in controls.iter() {
+            if child
+                .downcast::<HTMLElement>()
+                .map_or(false, |c| c.is_listed_element())
+            {
+                if child.has_attribute(&local_name!("id")) ||
```

Comment:
The specification text is ambiguous, but we actually want to verify that the id attribute's value equals name as well.

Reply:
Done.